lib/middleware/authentication/authentication.test.js

```diff
@@ -0,0 +1,145 @@
+/* eslint-env jest */
+// NPM dependencies
+jest.mock('basic-auth')
+jest.mock('express/lib/response')
+const basicAuth = require('basic-auth')
+
+// Local dependencies
+let authentication = require('./authentication.js')
+
+// Local variables
+const userDetails = {
+  name: 'secret-username',
+  pass: 'secure-password'
+}
+const next = jest.fn()
+
+const res = require('express/lib/response')
+
+// Mock console.log so we can check any output
+console.error = jest.fn()
+
+describe('authentication', () => {
+  it('should be a function', () => {
+    expect(authentication).toBeInstanceOf(Function)
+  })
+
+  describe('how it runs in prod', () => {
+    beforeAll(() => {
+      process.env.NODE_ENV = 'production'
+      process.env.USE_AUTH = 'true'
+    })
+
+    describe('server with no username/password set', () => {
+      beforeAll(() => {
+        delete process.env.USERNAME
+        delete process.env.PASSWORD
+      })
+
+      beforeEach(() => {
+        console.error.mockClear()
+        authentication({}, res, next)
+      })
+
+      it('should return a console error', () => {
+        expect(console.error.mock.calls[0][0]).toBe('Username or password is not set.')
+      })
+
+      it('should send marked up error message', () => {
+        expect(res.send.mock.calls[0][0]).toBe('<h1>Error:</h1><p>Username or password not set. <a href="https://govuk-prototype-kit.herokuapp.com/docs/publishing-on-heroku#6-set-a-username-and-password">See guidance for setting these</a>.</p>')
+      })
+    })
+
+    describe('server with username/password set', () => {
+      beforeAll(() => {
+        process.env.USERNAME = 'secret-username'
+        process.env.PASSWORD = 'secure-password'
+      })
+
+      beforeEach(() => {
+        jest.clearAllMocks()
+        basicAuth.mockReturnValue(userDetails)
+        authentication({}, res, next)
+      })
+
+      it('should not return a console error', () => {
+        expect(console.error).not.toBeCalled()
+      })
+
+      it('should call basic-auth and return username/password', () => {
+        expect(basicAuth).toReturnWith(userDetails)
+      })
+
+      it('should not set authentication header', () => {
+        expect(res.set).not.toBeCalled()
+      })
+
+      it('should not return 401/UnAuthorized', () => {
```

Comment:
this test seems to be whether `sendStatus` is called, should we change the name, or the test? Or is it fine as is?

Reply:
It's also inconsistent with line 101, which seems like a related test

Reply:
Good spot on line 101, the description is incorrect. It should read `should return 401/Authorized`